### PR TITLE
[Easy] emit tokenID on Trade/Revert event

### LINKED ISSUE #479: Emit buy/sell token in Trade/Reversion event
Request from Lev/Martin

This will allow knowing which tokens got traded at what price right from the event, rather than having to look up data at a historic block or index very historic events.

### PR DESCRIPTION
**Pull request by fleupold**
Closes #479

Unfortunately, there can only by three indexed fields on an event. It might be worth discussing if we should rather index both tokens and not index the orderId.

### TestPlan

CI

### CODE REVIEW

**Review by bh2smith — APPROVED** [review]
I am not sure what the benefit of indexing is here, but the change looks good to me.. Seems a little lop-sided to have sellToken "indexed" and not buyToken, but whatever.

**Comment by josojo** [comment]
I think that the indexing on the fields `owner` and `orderId` are the most important ones, as it will enable to see what happened to one particular order.

I would keep it as it is proposed in this PR.

**Review by josojo — APPROVED** [review]
Good to go

**Comment by fleupold** [comment]
I can add a comment saying that solidity Limits indexing to three fields to make it less clowny

File: contracts/BatchExchange.sol [hunk 1]
```diff
@@ -118,11 +118,27 @@ contract BatchExchange is EpochTokenLocker {
...
      */
-    event TradeReversion(address indexed owner, uint16 indexed orderId, uint128 executedSellAmount, uint128 executedBuyAmount);
+    event TradeReversion(
+        address indexed owner,
+        uint16 indexed orderId,
+        uint16 indexed sellToken,
+        // Solidity only supports three indexed arguments
+        uint16 buyToken,
+        uint128 executedSellAmount,
+        uint128 executedBuyAmount
```
Comment (nlordell) [inline contracts/BatchExchange.sol:140]: Ubernit+: Can we add a trailing comma here?

**Review by nlordell — APPROVED** [review]
LGTM, Is there a reason you prefer to index the sell token instead of the buy token?

File: contracts/BatchExchange.sol (hunk 1 above)
Comment (fleupold) [reply contracts/BatchExchange.sol:140]: We cannot:

> dex-contracts/contracts/BatchExchange.sol:128:34: ParserError: Unexpected trailing comma in parameter list

**Comment by fleupold** [comment]
> Is there a reason you prefer to index the sell token instead of the buy token?

Not really, it just came first in the list. Do you have an argument to suggest otherwise?

Note, the e2e test is failing since we are using more than 5.5M gas for the big ring now (5.53M gas)

**Comment by nlordell** [comment]
> Not really, it just came first in the list. Do you have an argument to suggest otherwise?

Nope, just curious!

File: contracts/BatchExchange.sol (hunk 1 above)
Comment (nlordell) [reply contracts/BatchExchange.sol:140]: Boo!

**Comment by bh2smith** [comment]
> Note, the e2e test is failing since we are using more than 5.5M gas for the big ring now (5.53M gas)

I intend to increase the gas limit in #485

**Pull request merged**